test/boost/database_test.cc

```diff
@@ -583,33 +598,54 @@ fs::path table_dir(const replica::column_family& cf) {
     return std::get<data_dictionary::storage_options::local>(cf.get_storage_options().value).dir;
 }
 
-SEASTAR_TEST_CASE(snapshot_works) {
-    return do_with_some_data({"cf"}, [] (cql_test_env& e) {
-        take_snapshot(e).get();
+static auto snapshot_works(const std::string& table_name) {
+    return do_with_some_data({"cf"}, [table_name] (cql_test_env& e) {
+        take_snapshot(e, "ks", table_name).get();
 
         std::set<sstring> expected = {
             "manifest.json",
+            "schema.cql"
         };
 
-        auto& cf = e.local_db().find_column_family("ks", "cf");
-        lister::scan_dir(table_dir(cf), lister::dir_entry_types::of<directory_entry_type::regular>(), [&expected] (fs::path parent_dir, directory_entry de) {
+        auto& cf = e.local_db().find_column_family("ks", table_name);
+        auto table_directory = table_dir(cf);
+        auto snapshot_dir = table_directory / sstables::snapshots_dir / "test";
+
+        lister::scan_dir(table_directory, lister::dir_entry_types::of<directory_entry_type::regular>(), [&expected](fs::path, directory_entry de) {
             expected.insert(de.name);
             return make_ready_future<>();
         }).get();
         // snapshot triggered a flush and wrote the data down.
-        BOOST_REQUIRE_GT(expected.size(), 1);
+        BOOST_REQUIRE_GE(expected.size(), 11);
 
         // all files were copied and manifest was generated
-        lister::scan_dir((table_dir(cf) / sstables::snapshots_dir / "test"), lister::dir_entry_types::of<directory_entry_type::regular>(), [&expected] (fs::path parent_dir, directory_entry de) {
+        lister::scan_dir(snapshot_dir, lister::dir_entry_types::of<directory_entry_type::regular>(), [&expected](fs::path, directory_entry de) {
             expected.erase(de.name);
             return make_ready_future<>();
         }).get();
 
         BOOST_REQUIRE_EQUAL(expected.size(), 0);
         return make_ready_future<>();
-    });
+    }, true);
 }
 
+SEASTAR_TEST_CASE(table_snapshot_works) {
+    return snapshot_works("cf");
+}
+
+SEASTAR_TEST_CASE(view_snapshot_works) {
+    return snapshot_works("view_cf");
+}
+
+SEASTAR_TEST_CASE(index_snapshot_works_1) {
+    return snapshot_works(::secondary_index::index_table_name("index_cf"));
+}
+
+SEASTAR_TEST_CASE(index_snapshot_works_2) {
+    return snapshot_works(::secondary_index::index_table_name("cf_c2_idx"));
+}
```

Comment:
Nitpick: why do you need to test this on two indexes? I'd just remove the second index and the index_snapshot_works_2 test. The fact that the default name for the index of column c2 on table cf is "cf_c2_inx" is something that is tested elsewhere, and not interesting to check in the context of snapshots. Or so I think.

Reply:
right, removed